pkg/apis/awsproviderconfig/v1alpha1/register.go

```diff
@@ -54,6 +57,15 @@ func NewScheme() (*runtime.Scheme, error) {
 	return SchemeBuilder.Build()
 }
 
+// Codec interface defines a minimal list that each machine codec needs to implement
+type Codec interface {
+	DecodeProviderConfig(*clusterv1.ProviderConfig, runtime.Object) error
+	DecodeProviderStatus(*runtime.RawExtension, runtime.Object) error
+	EncodeProviderStatus(runtime.Object) (*runtime.RawExtension, error)
+	ProviderStatusFromMachine(machine *clusterv1.Machine) (*AWSMachineProviderStatus, error)
```

Comment:
last two could arguably be actuator utils rather thant belong to this interface

Reply:
Removed both completely and kept only Decode/Encode methods for status/config.